src/main/scala/dpla/ingestion3/mappers/providers/PaExtractor.scala

```diff
@@ -3,57 +3,96 @@ package dpla.ingestion3.mappers.providers
+    if (contributors.nonEmpty)
+      nameOnlyAgent(contributors.last)
+    else
+      throw new Exception("Missing required property dataProvider because dc:contributor is empty")
+  }
+  // Get the second occurrence of the dc:identifier property
+  def itemUri(): ExactlyOne[URI] = {
+    val ids = extractStrings(xml \ "metadata" \\ "identifier")
+    if (ids.size >= 2)
+      new URI(ids(1))
+    else
+    // TODO Are these exception messages valid? Or is there a cleaner way of writing these?
+      throw new Exception("Missing required property itemUri because dc:identifier " +
+        s"does not occur at least twice in record: ${getProviderBaseId().get}")
+  }
+
+  def build: DplaMap = {
+    Try {
+      DplaMapData(
+        DplaSourceResource(
+          // This method of using NodeSeq is required because of namespace issues.
+          collection = extractStrings(xml \ "metadata" \\ "relation").headOption.map(nameOnlyCollection).toSeq,
+          contributor = extractStrings(xml \ "metadata" \\ "contributor").dropRight(1).map(nameOnlyAgent),
+          creator = extractStrings(xml \ "metadata" \\ "creator").map(nameOnlyAgent),
+          date = extractStrings(xml \ "metadata" \\ "date").map(stringOnlyTimeSpan),
+          description = extractStrings(xml \ "metadata" \\ "description"),
+          format = extractStrings(xml \ "metadata" \\ "type").filterNot(isDcmiType),
+          genre = extractStrings(xml \ "metadata" \\ "type").map(nameOnlyConcept),
+          identifier = extractStrings(xml \ "metadata" \\ "identifier"),
+          language = extractStrings(xml \ "metadata" \\ "language").map(nameOnlyConcept),
+          place = extractStrings(xml \ "metadata" \\ "coverage").map(nameOnlyPlace),
+          publisher = extractStrings(xml \ "metadata" \\ "publisher").map(nameOnlyAgent),
+          relation = extractStrings(xml \ "metadata" \\ "relation").drop(1).map(eitherStringOrUri),
+          rights = extractStrings(xml \ "metadata" \\ "rights"),
+          subject = extractStrings(xml \ "metadata" \\ "subject").map(nameOnlyConcept),
+          title = extractStrings(xml \ "metadata" \\ "title"),
+          `type` = extractStrings(xml \ "metadata" \\ "type").filter(isDcmiType)
+        ),
+
+        EdmWebResource(
+          // TODO is this the correct mapping for uri? What about OreAgg.`object`?
+          uri = itemUri,
+          fileFormat = extractStrings("dc:format")
+        ),
+
+        OreAggregation(
+          uri = mintDplaItemUri(),
+          //below will throw if not enough contributors
+          dataProvider = dataProvider(),
+          originalRecord = rawData,
+          provider = agent,
+          preview = thumbnail
+          // TODO what about `object` in this context? Assume no implementation
+        )
       )
-    )
+    } match {
+      case Success(s) => s
+        // TODO Is there a way to write this so that we don't have to write this match block into every Extractor?
```

Comment:
yeah, we could have an implicit function in scope that turns a Try[DplaMap] into either a DplaMapData or a DplaMapError, moving everything within match {} to that.